src/services/pegout-data.processor.ts

```diff
@@ -71,7 +78,49 @@ export class PegoutDataProcessor implements FilteredBridgeTransactionProcessor {
     return events.some(event => event.name === BRIDGE_EVENTS.RELEASE_REQUESTED);
   }
 
-  private async processReleaseRequestedStatus(extendedBridgeTx: ExtendedBridgeTx): Promise<void> {
+  private hasUpdateCollectionsEvent(events: BridgeEvent[]): boolean {
+    return events.some(event => event.name === BRIDGE_EVENTS.UPDATE_COLLECTIONS);
+  }
+
+  private async saveManyAsWaitingForSignature(pegouts: PegoutStatusDbDataModel[]) {
+    pegouts.forEach(async (pegout: PegoutStatusDbDataModel) => {
+      pegout.status = PegoutStatus.WAITING_FOR_SIGNATURE;
+      await this.pegoutStatusDataService.set(pegout);
+    });
+  }
+
+  private async processWaitingForSignaturesStatus(extendedBridgeTx: ExtendedBridgeTx): Promise<void> {
+    const currentBlockHeight = extendedBridgeTx.blockNumber;
+    this.logger.trace(`[processWaitingForSignaturesStatus] currentBlockHeight: ${currentBlockHeight}`);
+    const rskMaximumConfirmation = Number(process.env.RSK_MINIMUM_CONFIRMATION);
+    const dbPegoutsWaitingForConfirmations = await this.pegoutStatusDataService.getManyWaitingForConfirmationNewest();
+    this.logger.trace(`[processWaitingForSignaturesStatus] number of dbPegoutsWaitingForConfirmations: ${dbPegoutsWaitingForConfirmations.length}`);
+    const dbPegoutsWithEnoughConfirmations: PegoutStatusDbDataModel[] = dbPegoutsWaitingForConfirmations.filter(dbPegout => {
+      const blockHeightDiff = currentBlockHeight - dbPegout.rskBlockHeight;
+      return blockHeightDiff >= rskMaximumConfirmation;
+    });
+    this.logger.trace(`[processWaitingForSignaturesStatus] number of dbPegoutsWaitingForConfirmations with enough confirmations: ${dbPegoutsWithEnoughConfirmations.length}`);
+    const bridgeState = await this.bridgeService.getBridgeState();
+    const pegoutsWaitingForConfirmationMap = bridgeState.pegoutsWaitingForConfirmations
+        .reduce((accumulator, pegout) => accumulator.set(pegout.rskTxHash, pegout), new Map<string, PegoutWaitingConfirmation>());
+    if(pegoutsWaitingForConfirmationMap.size === 0) {
+      this.logger.trace(`[processWaitingForSignaturesStatus] no transactions in waiting for confirmation in the bridge state.`);
+      // If none of the pegouts in the db waiting for confirmation are found in the bridge state,
```

Comment:
We are only allowing TODOs in the comments. It seems to me that the default is to remove them.